src/nf/nf_activation.f90

```diff
@@ -25,12 +25,14 @@ module nf_activation
   contains
     procedure(eval_1d_i), deferred :: eval_1d
     procedure(eval_1d_i), deferred :: eval_1d_prime
+    procedure(eval_2d_i), deferred :: eval_2d
```

Comment:
Question to @milancurcic :
Do we really need 2D and 3D activation functions? I used 1D activations for 2D shapes without any issues and overcomplication of my code.
And, if we do need them, maybe use assumed-ranks here? I'm not really a fan of those copypasted generics

Reply:
I'll look into it more but I think what you noticed that you could do is a non-standard behavior provided by the compiler, and in that case, not portable.